src/litserve/server.py

```diff
@@ -256,17 +258,18 @@ def cleanup(request_buffer, uid):
 
 @asynccontextmanager
 async def lifespan(app: FastAPI):
-    app.request_queue = Queue()
     manager = Manager()
     app.request_buffer = manager.dict()
+    app.request_queue = manager.Queue()
 
     process_list = []
     # NOTE: device: str | List[str], the latter in the case a model needs more than one device to run
     for worker_id, device in enumerate(app.devices * app.workers_per_device):
         if len(device) == 1:
             device = device[0]
 
-        process = Process(
+        ctx = mp.get_context("spawn")
+        process = ctx.Process(
```

Comment:
the fundamental difference here is that we'll need `lit_api` to always be picklable if we go for `spawn`, since we pass it as an argument

now API should not have anything that can't be pickled, because the gnarly stuff typically happens in `setup`

if we go with spawn, can we detect the pickle exception on lit_api and tell users: this can't be pickled, whatever you do, do it in `setup`?

Reply:
we can do a `pickle.dumps` check before we start creating processes but haven't been able to find a way to detect after creating the process.

Reply:
yeah a check prior would work

Reply:
added the check @lantiga. could you take a look please?